core/src/main/scala/org/apache/spark/deploy/security/HadoopDelegationTokenManager.scala

```diff
@@ -42,7 +42,7 @@ import org.apache.spark.internal.Logging
 private[spark] class HadoopDelegationTokenManager(
     sparkConf: SparkConf,
     hadoopConf: Configuration,
-    fileSystems: Set[FileSystem])
+    fileSystems: () => Set[FileSystem])
```

Comment:
> the auth to visit FS is still "SIMPLE"

That sounds like your HDFS configuration is missing something. The `yarnConf` object passed to `hadoopFSsToAccess()` should have Kerberos configured as the auth mechanism, and because the container has valid delegation tokens, the HDFS library should use them.

Otherwise, I don't see how this feature would work at all; when submitting an app in cluster mode, the driver does not do a kerberos login (until tokens need to be renewed), and yet applications are able to talk to HDFS.